src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/router/street/AccessEgressRouter.java

```diff
@@ -23,34 +36,76 @@ public class AccessEgressRouter {
   private AccessEgressRouter() {}
 
   /**
-   * @param fromTarget whether to route from or towards the point provided in the routing request
-   *                   (access or egress)
-   * @return Transfer objects by access/egress stop
+   * Search for accesses or egresses. This will return both:
+   * <li> Directly accessible stops if the source/destination is a stop id.
+   * <li> Stops reachable by street search from the coordinates that correspond to the source/destination.
+   *
    */
-  public static Collection<NearbyStop> streetSearch(
+  public static Collection<NearbyStop> findAccessEgresses(
     RouteRequest request,
     TemporaryVerticesContainer verticesContainer,
     StreetRequest streetRequest,
-    DataOverlayContext dataOverlayContext,
-    boolean fromTarget,
+    @Nullable DataOverlayContext dataOverlayContext,
+    AccessEgressType accessOrEgress,
     Duration durationLimit,
     int maxStopCount
   ) {
     OTPRequestTimeoutException.checkForTimeout();
-    var nearbyStopFinder = new StreetNearbyStopFinder(
-      durationLimit,
-      maxStopCount,
-      dataOverlayContext
-    );
-    Collection<NearbyStop> nearbyStopList = nearbyStopFinder.findNearbyStops(
-      fromTarget ? verticesContainer.getToVertices() : verticesContainer.getFromVertices(),
-      fromTarget,
+
+    // Note: We do direct and street search in two parts since some stations will use the centroid
+    // for street routing, but should still give direct access/egresses to its child-stops.
+    var directAccessEgress = AccessEgressRouter.findDirectAccessEgress(
+      verticesContainer,
       request,
-      streetRequest
+      streetRequest,
+      accessOrEgress
     );
 
-    LOG.debug("Found {} {} stops", nearbyStopList.size(), fromTarget ? "egress" : "access");
+    // When looking for street accesses/egresses we ignore the already found direct accesses/egresses
+    var ignoreVertices = directAccessEgress
+      .stream()
+      .map(nearbyStop -> nearbyStop.state.getVertex())
+      .collect(Collectors.toSet());
+
+    var originVertices = accessOrEgress.isEgress()
+      ? verticesContainer.getToVertices()
+      : verticesContainer.getFromVertices();
+    var streetAccessEgress = new StreetNearbyStopFinder(
+      durationLimit,
+      maxStopCount,
+      dataOverlayContext,
+      ignoreVertices
+    )
+      .findNearbyStops(originVertices, request, streetRequest, accessOrEgress.isEgress());
+
+    var results = ListUtils.combine(directAccessEgress, streetAccessEgress);
+    LOG.debug("Found {} {} stops", results.size(), accessOrEgress);
+    return results;
+  }
 
-    return nearbyStopList;
+  /**
+   * Return a list of direct accesses/egresses that do not require any street search. This will
+   * return an empty list if the source/destination is not a stopId.
+   */
+  private static List<NearbyStop> findDirectAccessEgress(
```

Comment:
Yeah, I was really struggling to come up with a good name for this concept and direct access/egress was the best that I could find, even though it is a bit confusing since we already have "direct" as a concept.

These are access/egresses to the stops that correspond to the location you search from/to. The stops you have implicit access to without having to use the street network. They have 0 length and don't show up as legs in the itinerary. If you search from a stop id that stop would have a direct access. If you search from a station the child stops would have direct accesses. And if you search from a coordinate you wouldn't have any direct access/egresses.

I designed it so that it's only within the AccessEgressRouter that we need to make a distinction between these and the street access/egresses.

Other names that I could come up with are:
- free access/egress
- immediate access/egress
- instant access/egress

Or do you have any ideas for a better name?